src/org/safs/rest/service/models/consumers/SafsrestAdapter.groovy

```diff
@@ -149,7 +149,7 @@ class SafsrestAdapter extends ClientPOJOAdapter {
         def rootUrl = "${uri.scheme}://${uri.host}:${uri.port}"
         propertyProvider.rootUrl = rootUrl
         
-        if (userid && password) {
+        if (userid && password && (propertyProvider.authToken == null)) {
```

Comment:
@leiwang008 That is a very good question.  The access token is set for a specific credential.  RESTCAT is designed to hold the userid and the authToken in the propertyProvider.  So, we could store the propertyProvider's in a Map keyed off of userid.  (RESTCAT tries not to store passwords in heap memory, and I have been continuing that concern).  Or, we can change the propertyProvider in RESTCAT to hold the Map.   If we don't use a Map, then the access token would be obtained every time if they switched back and forth between userids.

Or we can require the user to create a different Service object for each userid.  I suppose the decision would be made based on user feedback and how difficult they would find it to create a separate Service object.  You would also be a better person than me to know which the user would find most convenient from the keyword testing point of view.